core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1166,17 +1165,25 @@ head = one
 
 -- | Returns the first element that satisfies the given predicate.
 --
--- @since 0.7.0
-{-# INLINE find #-}
-find :: Monad m => (a -> Bool) -> Fold m a (Maybe a)
-find predicate = foldt' step (Partial ()) (const Nothing)
+-- /Pre-release/
+{-# INLINE findM #-}
+findM :: Monad m => (a -> m Bool) -> Fold m a (Maybe a)
+findM predicate = Fold step (return $ Partial ()) (const $ return Nothing)
 
     where
 
     step () a =
-        if predicate a
-        then Done (Just a)
-        else Partial ()
+        (\r ->
+            if r
+            then Done (Just a)
+            else Partial ()) <$> predicate a
```

Comment:
Any reason for this style of code?
Maybe move the function to a where or a let statement?
This is rather hard to read.

Reply:
reason: least amount of change from previous code. Yes, can define the func separately.